lib/xapi-stdext-date/date.ml

```diff
@@ -62,20 +69,23 @@ let of_float x =
     time.Unix.tm_sec |> of_string
 
 (* Convert tm in localtime to calendar time, x *)
-let to_float_localtime x =
-  let datetime_to_float y mon d h min s =
-    fst Unix.(mktime { tm_year = y - 1900;
-                       tm_mon = mon - 1;
-                       tm_mday = d;
-                       tm_hour = h;
-                       tm_min = min;
-                       tm_sec = s;
-                       (* These are ignored: *)
-                       tm_wday = 0; tm_yday = 0; tm_isdst = true;
-                     })
-  in
-  let ((y, mon, d), ((h, min, s), _)) = Ptime.to_date_time x in
```

Comment:
Sounds like we should be able to use Ptime for this alone, without mixing Ptime with Unix.mktime. See Ptime_clock it has functions for determining the offset if that is what you need.
There is also `Ptime.to_float_s`.